### PR TITLE
feat(artifacts): start of artifact docs

### PR DESCRIPTION
**Pull request by lwander**
This kicks of the artifact documentation. 

Still missing and critical:
  1. Configuration (under setup),
  2. Common artifact formats (gcs vs. docker vs. s3...)
  3. Provider support

https://github.com/spinnaker/spinnaker/issues/2177

### CODE REVIEW

File: reference/artifacts/in-pipelines/index.md [hunk 1]
```diff
@@ -0,0 +1,137 @@
...
+{% include toc %}
+
+> :warning: Much of the behavior defined here depends on looking up execution
+> history within Redis. As a result, deleting recent executions from Redis is
+> undefined behavior.
+
+Once you have an idea of what an "artifact" is in Spinnaker, it's important to
+understand how they are used within pipelines. The idea is simple, artifacts
+arrive in a pipeline execution either from an external trigger or a stage, and
+are consumed by other stages based on pre-defined behavior. In order to explain
```
Comment (duftler) [inline reference/artifacts/in-pipelines/index.md]: Maybe: s/other/downstream

File: reference/artifacts/in-pipelines/index.md [hunk 2]
```diff
@@ -0,0 +1,137 @@
...
+how this works, we need to understand the concept of an "expected artifact".
+
+# Expected Artifacts
+
+Expected artifacts exist for two reasons:
+
+1. To declare what artifacts need to be present at a certain time during
```
Comment (duftler) [inline reference/artifacts/in-pipelines/index.md]: Maybe: s/time/point

File: reference/artifacts/in-pipelines/index.md [hunk 3]
```diff
@@ -0,0 +1,137 @@
...
+    "type": "docker/image"
+    {% endraw %}
+  }
+]
+```
+
+## Find Artifact from Execution
+
+To allow you to promote artifacts between executions, you can make use of the
+"Find Artifact from Execution" stage. All that's required is the ID of the
```
Comment (duftler) [inline reference/artifacts/in-pipelines/index.md]: Is this the execution ID or the pipeline name?

File: reference/artifacts/index.md [hunk 4]
```diff
@@ -0,0 +1,103 @@
...
+
+  "name":       // (Optional) A human-readable name that makes matching
+                // artifacts simpler.
+
+  "version":    // (Optional) The version of this artifact. By convention, the
+                // "version" should be compared against other artifacts with
+                // the same "type" and "name".
+
+  "provenance": // (Optional) A link to whatever produced this artifact. This
+                // used for deep-linking into other systems from Spinnaker.
```
Comment (duftler) [inline reference/artifacts/index.md]: Word or words missing between 'this' and 'used'.

**Review by duftler — COMMENTED** [review]
LGTM.
Please wait for @dorbin to review as well.

**Comment by dorbin** [comment]
I'm gonna go ahead and merge this as-is, and do an editing run in a separate PR rather than polluting it with lots of long comments.

File: reference/artifacts/in-pipelines/index.md (hunk 3 above)
Comment (lwander) [reply reference/artifacts/in-pipelines/index.md]: pipeline ID, clearing up the wording...

File: reference/artifacts/in-pipelines/index.md (hunk 2 above)
Comment (lwander) [reply reference/artifacts/in-pipelines/index.md]: :+1:

**Comment by lwander** [comment]
Thanks - I've addressed @duftler's feedback

**Pull request merged**